compiler_gym/envs/llvm/llvm_benchmark.py

```diff
@@ -422,3 +423,80 @@ def _add_path(path: Path):
     timestamp = datetime.now().strftime("%Y%m%HT%H%M%S")
     uri = f"benchmark://user-v0/{timestamp}-{random.randrange(16**4):04x}"
     return Benchmark.from_file_contents(uri, bitcode)
+
+
+def split_benchmark_by_function(
+    benchmark: Benchmark, timeout_seconds: float = 300
+) -> List[Benchmark]:
+    """Split a benchmark into a list of benchmarks, one per function."""
+    original_uri = deepcopy(benchmark.uri)
+    original_bitcode = benchmark.proto.file.contents
+
+    # Count the number of functions in the benchmark.
+    with Popen(
+        [str(llvm.llvm_extract_one_path()), "-", "-count-only", "-o", "-"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+    ) as p:
+        stdout, stderr = p.communicate(original_bitcode, timeout=timeout_seconds)
+        if p.returncode:
+            raise ValueError(
+                "Failed to count number of functions in benchmark: "
+                f"{stderr.decode('utf-8')}"
+            )
+    number_of_functions = int(stdout.decode("utf-8"))
+    if number_of_functions <= 0:
+        raise ValueError("No functions found!")
+
+    # Iterate over the number of functions, extracting each one in turn.
+    split_benchmarks: List[Benchmark] = []
+    for i in range(number_of_functions):
+        with Popen(
```

Comment:
I think you are missing `stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE`

Reply:
Good catch thanks!